loopy/tools.py

```diff
@@ -99,7 +99,8 @@ def update_for_dict(self, key_hash, key):
         unordered_hash(
             key_hash,
             (self.rec(self.new_hash(), (k, v)).digest()
-                for k, v in key.items()))
+                for k, v in key.items()),
+            hash_constructor=self.new_hash)
 
```

Comment:
could this code just be 
`update_for_dict=KeyBuilderBase.update_for_frozendict`?

Reply:
Indeed! Much simpler. Thanks!